scanapp/tasks.py

```diff
@@ -74,6 +80,55 @@ def scan_code_async(URL, scan_id, path):
         folder_name = None
         apply_scan_async.delay(path, scan_id, scan_type, URL, folder_name)
 
+
+@app.task
+def scan_code_async_final(URL, scan_id, scan_type):
+    """
+    Create and save a file at `path` present at `URL` using `scan_id` and bare `path`
+    and apply the scan.
+    """
+    logger.info('git repo detected')
+
+    dir_name = 'temp'
+
+    home_path = expanduser("~")
+
+    os.chdir(home_path)
+
+    if os.path.isdir(dir_name):
+        shutil.rmtree(dir_name)
```

Comment:
Are you serious? You are deleting a whole random directory tree? Use a temp directory instead!